### ISSUE TITLE
Mask in LGCP simulation may not be compatible with mean field

### ISSUE THREAD

**Issue by rubak** [spatstat/spatstat.random#2]
See [this post on StackOverflow](https://stackoverflow.com/q/78054924/3341769) for details.

It seems like the window mask for simulation is defined here without checking for compatibility with `mu` if it is supplied as an image:
  
https://github.com/spatstat/spatstat.random/blob/39802c1a5822db638cdd824661bcc23c3148b7ca/R/randomcircembed.R#L35

**Comment by baddstats** [comment]
That's true, but `rGRFcircembed` is internal and it assumes that this compatibility has already been checked. 

The warning about incompatible images comes from line 59 in `randomcircembed.R` where the zero-mean realisation is added to the mean field.

I will investigate further.

**Comment by baddstats** [comment]
The example provided by the user constructs an image object incorrectly, or at least, results in an image `mu` which is incompatible with the rectangle `win`. The code in `rLGCP` has detected but not automatically repaired this incompatibility, and I guess we could call that a bug. I have added a fix to `rLGCP` so that this should now be automatically repaired. I have responded to the StackOverflow post.
